crates/wasmtime/src/runtime/types/matching.rs

```diff
@@ -210,10 +210,7 @@ fn memory_ty(expected: &Memory, actual: &Memory, actual_runtime_size: Option<u64
 fn match_heap(expected: WasmHeapType, actual: WasmHeapType, desc: &str) -> Result<()> {
     use WasmHeapType as H;
     let result = match (actual, expected) {
-        (H::Concrete(actual), H::Concrete(expected)) => {
-            // TODO(dhil): we need either canonicalised types or a context here.
-            actual == expected
-        }
+        (H::Concrete(actual), H::Concrete(expected)) => actual == expected,
```

Comment:
I guess the original TODO, in the context of the typed function references proposal which did not introduce function type subtyping, is resolved since we have canonicalized types at this point now. But we have a new TODO for Wasm GC, which introduces function type subtyping, to do subtyping checks. I can add this second TODO.